### PR TITLE
#209: Input.NoNull, Ouput.NoNull and Func.NoNull

### PR DESCRIPTION
**Pull request by fabriciofx**
As per #209 .

`NotNullInput` moved to `Input`.
`NotNullOutput` moved to `Output`.
`Func.NoNulls` introduced.

### CODE REVIEW

**Comment by 0crat** [comment]
@fabriciofx I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/Func.java [hunk 1]
```diff
@@ -56,4 +56,40 @@
...
+                throw new IllegalArgumentException(
+                    "NULL instead of a valid function"
+                );
+            }
+            if (input == null) {
+                throw new IllegalArgumentException(
+                    "NULL instead of a valid input"
+                );
+            }
+            return this.func.apply(input);
```
Comment (yegor256) [inline src/main/java/org/cactoos/Func.java]: @fabriciofx we should check the result too

File: src/main/java/org/cactoos/Input.java [hunk 2]
```diff
@@ -60,4 +61,31 @@
...
+         */
+        public NoNulls(final Input input) {
+            this.origin = input;
+        }
+        @Override
+        public InputStream stream() throws IOException {
+            if (this.origin == null) {
+                throw new IOException("NULL instead of a valid input");
+            }
+            return this.origin.stream();
```
Comment (yegor256) [inline src/main/java/org/cactoos/Input.java]: @fabriciofx we should check the result too

File: src/main/java/org/cactoos/Output.java [hunk 3]
```diff
@@ -65,4 +65,31 @@
...
+         */
+        public NoNulls(final Output output) {
+            this.origin = output;
+        }
+        @Override
+        public OutputStream stream() throws IOException {
+            if (this.origin == null) {
+                throw new IOException("NULL instead of a valid output");
+            }
+            return this.origin.stream();
```
Comment (yegor256) [inline src/main/java/org/cactoos/Output.java]: @fabriciofx check the result

**Comment by yegor256** [comment]
@fabriciofx see above

File: src/main/java/org/cactoos/Func.java (hunk 1 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/Func.java]: @yegor256 Ok.

File: src/main/java/org/cactoos/Input.java (hunk 2 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/Input.java]: @yegor256 Ok.

File: src/main/java/org/cactoos/Output.java (hunk 3 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/Output.java]: @yegor256 Ok.

**Comment by fabriciofx** [comment]
@yegor256 Could you check it now?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**